workbench-core/environments-ui/src/pages/environments/new.tsx

```diff
@@ -0,0 +1,337 @@
+  const { projects, areProjectsLoading } = useProjects();
+  const { datasets, areDatasetsLoading } = useDatasets();
+
+  const breadcrumbs: BreadcrumbGroupProps.Item[] = [
+    {
+      text: 'Service Workbench',
+      href: '/'
+    },
+    {
+      text: 'Workspaces',
+      href: '/environments'
+    },
+    {
+      text: 'Create Workspace',
+      href: '/environments/new'
+    }
+  ];
+  const validationRules = [
+    {
+      field: 'name',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => !!a,
+      message: 'Workspace Name is Required'
+    },
+    {
+      field: 'name',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => !!a && nameRegex.test(a),
+      message:
+        'Workspace Name must start with an alphabetic character and can only contain alphanumeric characters (case sensitive) and hyphens.'
+    },
+    {
+      field: 'name',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => !!a && a.length <= 128,
+      message: 'Workspace Name cannot be longer than 128 characters'
+    },
+    {
+      field: 'projectId',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => !!a,
+      message: 'Project ID is Required'
+    },
+    {
+      field: 'envTypeId',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => !!a,
+      message: 'Compute Platform is Required'
+    },
+    {
+      field: 'envTypeConfigId',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => !!a,
+      message: 'Configuration is Required'
+    },
+    {
+      field: 'description',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => a && a.length <= 500,
+      message: 'Description cannot be longer than 500 characters'
+    },
+    {
+      field: 'description',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => !!a,
+      message: 'Description is Required'
+    }
+  ];
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const validateField = (field: keyof CreateEnvironmentForm, value: any): boolean => {
+    for (const rule of validationRules.filter((f) => f.field === field)) {
+      // eslint-disable-next-line security/detect-object-injection
+      if (!rule.condition(value)) {
+        setFormErrors((prevState: CreateEnvironmentFormValidation) => ({
+          ...prevState,
+          [`${field}Error`]: rule.message
+        }));
+        return false;
+      }
+    }
+    setFormErrors((prevState: CreateEnvironmentFormValidation) => ({ ...prevState, [`${field}Error`]: '' }));
+    return true;
+  };
+  const onSelectEnvType = async (selection: EnvTypeItem[]): Promise<void> => {
+    const selected = (selection && selection[0]) || undefined;
```

Comment:
```suggestion
    const selected = selection[0];
```

Reply:
since selection must be an array, this would work since `selection[0]` would return undefined for an empty array

Reply:
This doesn't work